### PR TITLE
Implement Clone for parse::Parser

### LINKED ISSUE #169: enable Clone for Parser?
Could we enable `Clone` for `Parser`? The use case is that I'd like in [display-as](https://crates.io/crates/display-as) to implement an optional `DisplayAs` integration with `Parser`, so users (myself, really) can have their markdown converted to HTML in the template generation.  Currently, I allocate a separate string for the markdown-generated HTML which is a bit inefficient and ugly.  The challenge is that `DisplayAs` takes a `&self`, which doesn't work with an iterator.  If I could clone the iterator, I could just mutate a copy of the object.

The other approach I could use would be to create a wrapper around a String (or `&str`) and then parse it while displaying it, but that would limit the use of `Option`s to customize the generation, so I wouldn't want to put that directly into `display-as`.

### PR DESCRIPTION
**Pull request by joshleeb**
This PR implements `Clone` for `parse::Parser`

Resolves #169

### CODE REVIEW

**Comment by joshleeb** [comment]
[Linking to comment in Issue](https://github.com/raphlinus/pulldown-cmark/issues/169#issuecomment-474218459)

> I've put up a PR that implements Clone for Parser, just to see how much work it would take.

Happy to close this PR if requiring the data within a tree `Node` implement `Clone` is not something that is desirable.

File: src/tree.rs [hunk 1]
```diff
@@ -55,21 +55,22 @@ impl Sub<usize> for TreeIndex {
     }
 }
 
-#[derive(Debug)]
-pub struct Node<T> {
+#[derive(Debug, Clone)]
+pub struct Node<T: Clone> {
```
[the remark this answers is not in the document]
Comment (oberien) [reply src/tree.rs]: That trait bound is automatically generated by the `derive(Clone)` implementation: <https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9df62839230042fb0b21a9bb099ab406>

[the remark this answers is not in the document]
Comment (marcusklaas) [reply src/tree.rs]: Cool, TIL! Then removing the bound should be enough here.

[the remark this answers is not in the document]
Comment (joshleeb) [reply src/tree.rs]: I learnt something here as well! 😄 Will update the trait bounds :+1:

**Comment by marcusklaas** [comment]
:+1: thanks!

**Pull request merged**